### ISSUE TITLE
Question - IIP Server and Openseadragon

### ISSUE THREAD

**Issue by terenceingram** [openseadragon/openseadragon#633]
Has anyone enabled openseadragon to interface with IIP Server? 
I currently load three image sizes using the Legacy Image Pyramids but am wondering if there is a better way?

**Comment by boskar** [comment]
I use it, works pretty good and fast, but You need to prepare pyramid tiff for the IIPServer.

**Comment by jpstroop** [comment]
If you're looking to save space, JPEG2000 will give you more resolution and
tile size options and use significantly less stage space than TIFFs.

I think you need a license for Kakadu to use JP2 with IIP, however.

You might also consider using the IIIF protocol with OSd, which will allow
you to use an number of different servers.

**Comment by terenceingram** [comment]
@boskar so I have to keep using the Legacy Image Pyramids to serve up my images? What about DZI? 
@jpstroop I have it all wired up using Legacy Image Pyramids and asking the IIP Server for 3 different image sizes. It seems to work but when I have seriously wide images. The original may be 2m wide. When I serve them up using the current method it pixelates when zooming in really deep. So I was hoping to use another method to put the images into the viewer.

**Comment by terenceingram** [comment]
@boskar @jpstroop 
Here's an example: http://nladom-test.nla.gov.au/nla.obj-41292842/view

**Comment by jpstroop** [comment]
@terenceingram Not sure that I completely understand your problem, but for whatever we've not seen any problems with wide images using JP2 + Loris/IIIF + OSd: http://pudl.princeton.edu/sheetreader.php?obj=5m60qs654

**Comment by terenceingram** [comment]
@jpstroop sorry the example I have is not publicly accessible. Your example is exactly what I am trying to achieve. So what is OSd and Loris/IIF? As the IIP Server is imbedded can I use it?

**Comment by jpstroop** [comment]
@terenceingram [This is Loris](https://github.com/loris-imageserver/loris) (Full discolsure: I'm the primary developer), which is another image server, written in Python, which also uses Kakadu to serve JP2 images. Loris and IIP both expose the [IIIF Image API](http://iiif.io/api/image/2.0/), which is gaining popularity with cultural heritage institutions like yours (and mine) and works natively with OpenSeadragon.

My sense is that it's the pyramid TIFFs that are causing your problem, but I'm afraid I don't have enough experience with those to offer advice. I think @terrellt might work with them + IIP + OSd, so maybe he can weigh in with something obvious.

**Comment by terrellt** [comment]
I don't know of anything obvious - we use VIPS to generate pyramidal tiffs via the documentation at http://iipimage.sourceforge.net/documentation/images/ and output it using IIP's DeepZoom API, which works great - certainly better than legacy image pyramids.

**Comment by terenceingram** [comment]
Everyone, just a small update. IIP - http://iipimage.sourceforge.net/documentation/protocol/ can provide Deep Zoom. I managed to wire an example up recently and my pixellation issue disappeared. My next plan is to change our use from Legacy Image Pyramids to using dzi from IIP. Hopefully I'll have time to do that in the next week or so.

**Comment by terenceingram** [comment]
@boskar @jpstroop Can you update the page about openseadragon in the wild with our production link?

http://nla.gov.au/nla.obj-41292850

We are currently migrating around 200 000 our of digitised images to the new platform and they will be viewable using openseadragon by the end of the month.

**Comment by iangilman** [comment]
@terenceingram Wow! That's quite a lot of images! 

So on the "In the wild" page, we're currently linking to the main page at http://trove.nla.gov.au/ and the detail page at http://nladom-test.nla.gov.au/nla.obj-41292842/view. Is it just the detail page link that needs updating?

**Comment by terenceingram** [comment]
@iangilman Yes just the detail link. All the images will be indexed in Trove (One of our discovery applications). We are also migrating thousands of digitised maps that will use openseadragon too. That's later in the year. We plan to have a view different ways to view our images and navigate the complex hierarchies. Openseadragon is used for Maps, Pictures, Ephemera, Manuscript (letters) and then we use the openarchive bookreader for our digitised books.

**Comment by iangilman** [comment]
Very cool! 

Ok, I've updated the link. Thanks for the heads up.

**Comment by terenceingram** [comment]
@iangilman We have migrated our images collection, digitised books and digitised sheet music to the new application. We are using a combination of openseadragon and the internet archive bookreader. But there would at least be over 300 000 images delivered via deep zoom using IIP to the openseadragon viewer. We will migrate our maps collection by early December. We have added a few more navigation features for our maps collection. Not sure the number of images but I suspect it will be at least 50 000. We are also building an openseadragon modal popup from our internet archive bookreader to load all the pages of the book into an openseadragon viewer to allow users better zooming. This is useful where a book contains a map or is some kind of manuscript.
